runtime/onert/core/include/ir/Operation.h

```diff
@@ -28,7 +28,10 @@ namespace onert
 namespace ir
 {
 
-class Operation : public IOperation
+// NOTE Virtual inheritance is a c++ technique that can be used to avoid the diamond problem by
+//      clarifying ambiguity over which ancestor class to use.
+//      Classes for trainable operations will inherit this class with the class ITrainableOperation.
```

Comment:
```suggestion
// NOTE Virtual inheritance is introduced because `Loss` inherits
//      `ITrainableOperation` and `Operation` which inherits `IOperation`).
```

Reply:
```suggestion
// NOTE Virtual inheritance is introduced because trainable operations inherits
//      `ITrainableOperation` and `Operation` which inherits `IOperation`).
```